library/core/src/array/mod.rs

```diff
@@ -440,32 +440,23 @@ impl<T: Copy> SpecArrayClone for T {
     }
 }
 
-// The Default impls cannot be done with const generics because `[T; 0]` doesn't
```

Comment:
Also this comment seems to be incorrect evident by the test `empty_array_is_always_default()` in `library/core/tests/array.rs`